lisa/sut_orchestrator/azure/common.py

```diff
@@ -1158,6 +1168,42 @@ def get_vm(platform: "AzurePlatform", node: Node) -> Any:
     return vm
 
 
+def get_nic_name(nic_resource_id: str) -> str:
+    nic_name_pattern = re.compile(r"Microsoft.Network/networkInterfaces/(.*)", re.M)
```

Comment:
always move re.compile out of method, so it doesn't need to be compiled every time, when calling the method.

Reply:
updated